romanesco/plugins/girder_io/__init__.py

```diff
@@ -44,27 +48,37 @@ def fetch_handler(spec, **kwargs):
     else:
         raise Exception('Invalid resource type: ' + resource_type)
 
-    return dest
+    if target == 'filepath':
+        return dest
+    elif target == 'memory':
+        with open(dest, 'rb') as fd:
+            return fd.read()
+    else:
+        raise Exception('Invalid Girder push target: ' + target)
 
 
 def push_handler(data, spec, **kwargs):
     parent_type = spec.get('parent_type', 'folder')
-    name = spec.get('name', os.path.basename(data))
-    description = spec.get('description')
+    taskOutput = kwargs.get('task_output', {})
+    target = taskOutput.get('target', 'filepath')
```

Comment:
I'll add something to the docs. I'd love for the default to be consistent, but changing it here would break existing code that uses the girder IO mode.